Source/ASDisplayNode+Yoga.mm

```diff
@@ -205,104 +197,59 @@ - (void)setYogaMeasureFuncIfNeeded
   // Size calculation via calculateSizeThatFits: or layoutSpecThatFits:
   // This will be used for ASTextNode, as well as any other node that has no Yoga children
   if (self.yogaChildren.count == 0) {
-    YGNodeRef yogaNode = self.yogaNode; // Use property to assign Ref if needed.
-    YGNodeSetContext(yogaNode, (__bridge void *)self);
-    YGNodeSetMeasureFunc(yogaNode, &ASLayoutElementYogaMeasureFunc);
+    // TODO(appleguy): Add override detection for calculateSizeThatFits: and calculateLayoutThatFits:,
+    // then we can set the MeasureFunc only for nodes that override one of the trio of measurement methods.
+    // if (_layoutSpecBlock == NULL && (_methodOverrides & ASDisplayNodeMethodOverrideLayoutSpecThatFits) == 0 && ...) {
+
+    YGNodeRef yogaNode = self.style.yogaNode;
+    ASDisplayNodeAssert(yogaNode, @"-[ASDisplayNode setYogaMeasureFuncIfNeeded] called with a NULL yogaNode! %@", self);
+    if (yogaNode) {
+      // TODO(appleguy): Use __bridge_transfer to retain nodes and clean up appropriately.
+      YGNodeSetContext(yogaNode, (__bridge void *)self);
```

Comment:
Maybe it is appropriate to use ASWeakProxy here, and strong-retain the proxy itself? Otherwise it seems a retain cycle would be created.